presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/SimplifyExpressions.java

```diff
@@ -271,4 +278,75 @@ public Expression rewriteLogicalBinaryExpression(LogicalBinaryExpression node, N
+        private final Map<Symbol, Type> typeAssignments;
+
+        public RemoveIdentityCastContext(Map<Symbol, Expression> expressionAssignments,
+                                         Map<Symbol, Type> typeAssignments)
+        {
+            this.expressionAssignments = expressionAssignments;
+            this.typeAssignments = typeAssignments;
+        }
+
+        public Map<Symbol, Expression> getExpressionAssignments()
+        {
+            return expressionAssignments;
+        }
+
+        public Map<Symbol, Type> getTypeAssignments()
+        {
+            return typeAssignments;
+        }
+    }
+
+    private static class RemoveIdentityCastsRewriter
+            extends ExpressionRewriter<RemoveIdentityCastContext>
+    {
+        @Override
+        public Expression rewriteExpression(Expression node, RemoveIdentityCastContext context,
+                                            ExpressionTreeRewriter<RemoveIdentityCastContext> treeRewriter)
+        {
+            if (context.getExpressionAssignments() == null || context.getTypeAssignments() == null) {
+                return node;
+            }
+            else {
+                Map<Symbol, Type> typeAssignments = context.getTypeAssignments();
+                for (Map.Entry<Symbol, Expression> expressionAssignment : context.getExpressionAssignments().entrySet()) {
+                    Symbol assignmentSymbol = expressionAssignment.getKey();
+                    Expression expression = expressionAssignment.getValue();
+                    if (expression == node) {
+                        if (expression instanceof Cast) {
+                            Expression expressionToCast = ((Cast) expression).getExpression();
+                            TypeSignature typeOfExpressionToCastTo = typeAssignments.get(assignmentSymbol).getTypeSignature();
+                            TypeSignature typeOfExpressionToCast;
+                            if (expressionToCast instanceof SymbolReference) {
+                                Symbol expressionSymbol = new Symbol(((SymbolReference) expressionToCast).getName());
+                                typeOfExpressionToCast = typeAssignments.get(expressionSymbol).getTypeSignature();
+                            }
+                            else if (expressionToCast instanceof GenericLiteral) {
+                                typeOfExpressionToCast = new TypeSignature(((GenericLiteral) expressionToCast).getType());
+                            }
+                            else {
+                                return expression;
+                            }
+
+                            if (typeOfExpressionToCast.equals(typeOfExpressionToCastTo)) {
+                                return expressionToCast;
+                            }
+                            else {
+                                return expression;
+                            }
+                        }
+                        else {
```

Comment:
Won't be needed after applying https://github.com/Teradata/presto/pull/329/commits/17a508110880b64c069957c8585b7d5f5653a291#r76855699